File: app/src/main/java/ru/ulstu_team/ulstuschedule/util/GsonUtil.java
package ru.ulstu_team.ulstuschedule.util;

import com.google.gson.ExclusionStrategy;
import com.google.gson.FieldAttributes;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import io.realm.RealmObject;

public class GsonUtil {

    private static Gson mGson;

    private GsonUtil() { }

    public static Gson getGsonInstance() {
        if (mGson == null) {
            mGson = new GsonBuilder()
                    .setExclusionStrategies(new ExclusionStrategy() {
                        @Override
                        public boolean shouldSkipField(FieldAttributes f) {
                            return f.getDeclaringClass().equals(RealmObject.class);
                        }

                        @Override
                        public boolean shouldSkipClass(Class<?> clazz) {
                            return false;
                        }
                    })
                    .create();
        }
        return mGson;
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/util/GsonUtil.kt
package ru.ulstu_team.ulstuschedule.util

import com.google.gson.ExclusionStrategy
import com.google.gson.FieldAttributes
import com.google.gson.Gson
import com.google.gson.GsonBuilder

import io.realm.RealmObject

object GsonUtil {

    private var mGson: Gson? = null

    val gsonInstance: Gson
        get() {
            if (mGson == null) {
                mGson = GsonBuilder().setExclusionStrategies(object : ExclusionStrategy {
                    override fun shouldSkipField(f: FieldAttributes): Boolean {
                        return f.declaringClass == RealmObject::class.java
                    }

                    override fun shouldSkipClass(clazz: Class<*>): Boolean {
                        return false
                    }
                }).create()
            }
            return mGson!!
        }
}

